### DISCUSSION TITLE
Exec example but using WithAltScreen() prints output to outside of the altscreen

### DISCUSSION THREAD

**Discussion by zd4y** [charmbracelet/bubbletea#424] [Q&A] [answered]
Hello, I want to have something similar to the exec example, but having the altscreen active from the beginning, like this (I only changed `main`):

```go
package main

import (
	"fmt"
	"os"
	"os/exec"

	tea "github.com/charmbracelet/bubbletea"
)

type editorFinishedMsg struct{ err error }

func openEditor() tea.Cmd {
	editor := os.Getenv("EDITOR")
	if editor == "" {
		editor = "vim"
	}
	c := exec.Command(editor) //nolint:gosec
	return tea.ExecProcess(c, func(err error) tea.Msg {
		return editorFinishedMsg{err}
	})
}

type model struct {
	altscreenActive bool
	err             error
}

func (m model) Init() tea.Cmd {
	return nil
}

func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
	switch msg := msg.(type) {
	case tea.KeyMsg:
		switch msg.String() {
		case "a":
			m.altscreenActive = !m.altscreenActive
			cmd := tea.EnterAltScreen
			if !m.altscreenActive {
				cmd = tea.ExitAltScreen
			}
			return m, cmd
		case "e":
			return m, openEditor()
		case "ctrl+c", "q":
			return m, tea.Quit
		}
	case editorFinishedMsg:
		if msg.err != nil {
			m.err = msg.err
			return m, tea.Quit
		}
	}
	return m, nil
}

func (m model) View() string {
	if m.err != nil {
		return "Error: " + m.err.Error() + "\n"
	}
	return "Press 'e' to open your EDITOR.\nPress 'a' to toggle the altscreen\nPress 'q' to quit.\n"
}

func main() {
	m := model{altscreenActive: true}
	if err := tea.NewProgram(m, tea.WithAltScreen()).Start(); err != nil {
		fmt.Println("Error running program:", err)
		os.Exit(1)
	}
}
```
However, when I run this program and press `e` to open the editor, sometimes, after exiting, the message
```
Press 'e' to open your EDITOR.
Press 'a' to toggle the altscreen
Press 'q' to quit.
```
gets printed outside of the altscreen. And also, just after pressing `e` you can see the terminal outside of the altscreen for a moment before the editor opens.

How can I fix this so that after exiting nothing gets printed?

**Comment by bashbunni** [comment, marked as answer]
Hey @zd4y, 
you can check out the Gist I made with the functionality you're looking for [here](https://gist.github.com/bashbunni/e3306e8633512d8134012028288212db).

Basically what you're looking to do is have a field in your model for when you're quitting your application so that `View` can output an empty string as the final render to the terminal. In this case with the editor, you have to temporarily set this as we have to exit the alt screen when the editor opens, then when we quit the editor we want to see the prompt again. You'll see all of that written as code in the example provided. Let me know if this answers your question!

**Reply by zd4y** [reply]
Thank you, this is much better than the original behavior. It would be great if the terminal was left unchanged after running the program though, now it clears the terminal after exiting. But this works, thanks :)

**Reply by meowgorithm** [reply]
Reading this all again, we may want to open the original issue as it sounds like it may be a bug.

**Reply by bashbunni** [reply]
@meowgorithm I think it might be a bug with opening the editor. It seems to do a "final render" when it exits the alt screen for the editor to open up. I'd imagine this would be the case for any `tea.ExecProcess`. I'd agree with re-opening the ticket

**Reply by bernardo1r** [reply]
@bashbunni Can you explain why if you toggle the altscreen in and out and terminate the program in it, the ```View``` does not clean the non altscreen buffer?

Here's a quick gif I made to show the behaviour:
![example](https://user-images.githubusercontent.com/43327473/202099218-9b8b1dbc-d613-4bc1-be97-85b84de297f3.gif)
